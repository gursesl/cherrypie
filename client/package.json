{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "apollo-cache-inmemory": "^1.1.3",
    "apollo-client": "^2.1.0",
    "apollo-link": "^1.0.5",
    "apollo-link-context": "^1.0.2",
    "apollo-link-error": "^1.0.2",
    "apollo-link-http": "^1.3.0",
    "apollo-link-ws": "^1.0.2",
    "apollo-utilities": "^1.0.3",
    "eslint-config-airbnb": "^16.1.0",
    "graphql": "^0.11.7",
    "graphql-tag": "^2.6.0",
    "history": "^4.7.2",
    "immutable": "^3.8.2",
    "localforage": "^1.5.5",
    "lodash": "^4.17.4",
    "prop-types": "^15.6.0",
    "react": "^16.2.0",
    "react-apollo": "^2.0.1",
    "react-dom": "^16.2.0",
    "react-immutable-proptypes": "^2.1.0",
    "react-redux": "^5.0.6",
    "react-router-dom": "^4.2.2",
    "react-router-redux": "^5.0.0-alpha.8",
    "react-scripts": "1.0.17",
    "redux": "^3.7.2",
    "redux-form": "^7.2.0",
    "redux-immutable": "^4.0.0",
    "redux-saga": "^0.16.0",
    "reselect": "^3.0.1",
    "semantic-ui-css": "^2.2.12",
    "semantic-ui-react": "^0.77.0",
    "styled-components": "^2.2.4",
    "subscriptions-transport-ws": "^0.9.4",
    "validator": "^9.1.2",
    "whatwg-fetch": "^2.0.3"
  },
  "scripts": {
    "clean": "rm -rf node_modules build && yarn install",
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "NODE_ENV=test ENABLE_MOCK_API=true MOCK_API_URL=http://localhost:3001 react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  },
  "devDependencies": {
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.0",
    "redux-mock-store": "^1.3.0",
    "redux-saga-testing": "^1.0.5",
    "ws": "^3.3.2"
  }
}
