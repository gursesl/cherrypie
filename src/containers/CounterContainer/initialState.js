import { fromJS } from 'immutable'

const initialState = fromJS({
  value: 0,
})

export default initialState
