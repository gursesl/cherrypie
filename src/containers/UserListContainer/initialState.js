import { fromJS } from 'immutable'
import { SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './constants'

const initialState = fromJS({
  [SELECTOR_USERS_USERS]: [
    {
      id: "95617189",
      firstName: "Elfrieda",
      lastName: "Frank",
      email: "Ada20@hotmail.com",
    },
    {
      id: "95617188",
      firstName: "Jim",
      lastName: "Smith",
      email: "jsmith@mail.com",
    },
  ],
  [SELECTOR_USERS_ERROR]: 'This is an error',
  [SELECTOR_USERS_ISLOADING]: true,
})

export default initialState
