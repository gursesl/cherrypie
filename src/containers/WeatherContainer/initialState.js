import { fromJS } from 'immutable'
import {
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from './constants'

const initialState = fromJS({
  [SELECTOR_WEATHER_ZIP]: '20165',
  [SELECTOR_WEATHER_CITY]: 'Sterling',
  [SELECTOR_WEATHER_RESULTS]: [],
  [SELECTOR_WEATHER_ERROR]: 'Load error',
})

export default initialState
