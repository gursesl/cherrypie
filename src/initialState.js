export default {
  users: [
    {
      id: "95617189",
      firstName: "Elfrieda",
      lastName: "Frank",
      email: "Ada20@hotmail.com"
    },
    {
      id: "95617188",
      firstName: "Jim",
      lastName: "Smith",
      email: "jsmith@mail.com"
    },
  ],
  error: null,
  isLoading: true,
  value: 10
}
