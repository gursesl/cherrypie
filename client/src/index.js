import 'semantic-ui-css/semantic.min.css'
// import './dist/semantic.min.css'
import React from 'react'
import { render } from 'react-dom'
import { Provider } from 'react-redux'
import { ApolloProvider } from 'react-apollo'
import { ConnectedRouter } from 'react-router-redux'
import store, { history } from './store'
import './index.css'
import App from './App'
import client from './apolloClient'
import registerServiceWorker from './registerServiceWorker'

const target = document.getElementById('root')

render(
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  </ApolloProvider>,
  target
)
registerServiceWorker()
