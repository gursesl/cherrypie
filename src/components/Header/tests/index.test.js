// REFERENCE: COMPONENT TEST WITH REACT-ROUTER 4
import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
// Apollo client
import { ApolloProvider } from 'react-apollo'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Immutable from 'immutable'
import configureStore from 'redux-mock-store'
import createHistory from 'history/createBrowserHistory'
import { Menu, MenuItem } from 'semantic-ui-react'
import client from '../../../apolloClient'
import '../../../setupTests'
import initialState from '../../../initialState'
import AppHeader from '../'

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore(Immutable.fromJS(initialState))
const history = createHistory()
const component = (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <AppHeader />
      </ConnectedRouter>
    </Provider>
  </ApolloProvider>
)

const props = {
  data: {
    loading: false,
    getCurrentUser: {
      id: 123,
      email: 'email@email.com',
      fullName: 'John Smith',
    },
  },
}

describe('AppHeader:index', () => {
  const shallowComponent = shallow(<AppHeader.WrappedComponent {...props} />)
  const deepComponent = mount(component)

  beforeEach(() => {})

  it('should render without blowing up', () => {
    expect(shallowComponent).toBeDefined()
  })

  it('should have the proper initial state', () => {
    expect(shallowComponent.instance().state).toEqual({ visible: false })
  })

  it('should render a Visibility element', () => {
    expect(deepComponent.find(Menu).length).toBe(1)
  })

  it('should have a predefined number of nested menu items', () => {
    expect(deepComponent.find(MenuItem).length).toBe(8)
  })

  it('should render a child <FixedMenu /> component', () => {
    expect(shallowComponent.find('FixedMenu').exists()).toBe(false)
  })

  it('renders itself into the DOM with the correct shallowComponent styles', () => {
    const wrapper = renderer.create(component)
    expect(wrapper).toMatchSnapshot()
  })

  // TODO: Implement scroll simulation and assert FixedMenu found
  it('should have FixedMenu visible if state visible parameter is set to true', () => {
    shallowComponent.instance().hideFixedMenu()
    expect(shallowComponent.instance().state.visible).toBeFalsy()
  })

  it('should have FixedMenu hidden if state visible parameter is set to false', () => {
    shallowComponent.instance().showFixedMenu()
    expect(shallowComponent.instance().state.visible).toBeTruthy()
  })
})
