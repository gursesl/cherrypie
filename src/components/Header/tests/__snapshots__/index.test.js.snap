// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader:index renders itself into the DOM with the correct shallowComponent styles 1`] = `
<div>
  <div>
    <div
      className="ui inverted vertical center aligned segment"
    >
      <div
        className="ui container"
      >
        <div
          className="ui large inverted pointing secondary menu"
        >
          <a
            aria-current={false}
            className="item"
            href="/"
            onClick={[Function]}
            style={undefined}
          >
            Home
          </a>
          <a
            aria-current={false}
            className="item"
            href="/users"
            onClick={[Function]}
            style={undefined}
          >
            Users
          </a>
          <a
            aria-current={false}
            className="item"
            href="/gqlusers"
            onClick={[Function]}
            style={undefined}
          >
            GQL Users
          </a>
          <a
            aria-current={false}
            className="item"
            href="/counter"
            onClick={[Function]}
            style={undefined}
          >
            Counter
          </a>
          <a
            aria-current={false}
            className="item"
            href="/weather"
            onClick={[Function]}
            style={undefined}
          >
            Weather
          </a>
          <div
            className="item"
            onClick={[Function]}
          >
            <a
              href="/profile"
              onClick={[Function]}
            >
              Profile
            </a>
          </div>
          <div
            className="item"
            onClick={[Function]}
          >
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </div>
          <div
            className="item"
            onClick={[Function]}
          >
            <a
              href="/logout"
              onClick={[Function]}
            >
              Logout
            </a>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
